js/src/Ice/ConnectionI.js

```diff
@@ -431,6 +439,8 @@ export class ConnectionI {
 
             this.sendMessage(OutgoingMessage.createForStream(os, true));
 
+            --this._dispatchCount;
```

Comment:
You also need it in sendNoResponse (= oneway dispatch that completes successfully).

In other languages, sendResponse and sendNoResponse were merged as sendResponse as part of the dispatch pipeline refactoring.